src/core/exec-invoke.c

```diff
@@ -901,8 +901,14 @@ static int get_supplementary_groups(
         bool keep_groups = false;
         if (user && gid_is_valid(gid) && gid != 0) {
                 /* First step, initialize groups from /etc/groups */
-                if (initgroups(user, gid) < 0)
-                        return -errno;
+                if (initgroups(user, gid) < 0) {
+                        /* If our primary gid is already the one specified in Group= (i.e. we're running in
+                         * user mode), gracefully handle the case where we have no privilege to re-initgroups().
+                         * The user's group memberships might have been modified, but the change hasn't been
+                         * applied to the the user manager through re-login. */
+                        if (!(ERRNO_IS_PRIVILEGE(errno) && gid == getgid()))
```

Comment:
Not a matter but I'd like to replace `!(A && B)` with `!A || !B` to reduce parentheses.